File: java/air-io/src/main/java/feuyeux/io/nio/udp/protocol/MouseInputMessage.java
package feuyeux.io.nio.udp.protocol;

public class MouseInputMessage extends BaseMessage {

    public static final int MOUSE_ACTION_MOVE_CURSOR = 0;
    public static final int MOUSE_ACTION_BUTTON_PRESS = 1;

    /*
     * For cursor move action: arg1=dx, arg2=dy
     * For button click action: arg1 and arg2 are not checked
     */
    private int mouseAction;
    private int arg1;
    private int arg2;

    public MouseInputMessage(int action, int arg1, int arg2) {
        cmdType = CMD_TYPE_MOUSE_INPUT;
        this.mouseAction = action;
        this.arg1 = arg1;
        this.arg2 = arg2;
    }

    public int action() {
        return mouseAction;
    }

    public int arg1() {
        return arg1;
    }

    public int arg2() {
        return arg2;
    }

    @Override
    public String toString() {
        return "[MouseInput] TYPE=" + cmdType + ",X=" + arg1 + ",Y=" + arg2 + ",MOUSE-ACTION=" + mouseAction;
    }
}

File: ios/learning/UdpCmd/UdpCmd/MouseInputMessage.m
#import "MouseInputMessage.h"

@implementation MouseInputMessage
- (NSString *)toString {
    return [NSString localizedStringWithFormat:@"TYPE=%d,ARG1=%d,ARG2=%d", self.cmdType, _arg1, _arg2];
}

- (MouseInputMessage *)init:(int)action arg1:(int)arg1 arg2:(int)arg2 {
    self = [super init];
    self.cmdType = MOUSE;
    [self setMouseAction:action];
    [self setArg1:arg1];
    [self setArg2:arg2];
    return self;
}
@end
